.github/workflows/github-runner-provisioner.yaml

```diff
@@ -23,7 +23,7 @@ jobs:
       - uses: actions/checkout@v4
       - uses: actions/setup-go@v4
         with:
-          go-version: '1.18'
+          go-version: '1.22'
```

Comment:
Should this be read from go.mod?
https://github.com/actions/setup-go?tab=readme-ov-file#getting-go-version-from-the-gomod-file

Reply:
It's better practice these days, but in this case the `go.mod` is buried under the `github-runner-provisioner` folder (which it shouldn't be, or at least it should still have a root go.mod). And at that point it's better to just use the version instead of hardcoding an inappropriate path. Presumably something like `go-version-file: github-runner-provisioner/go.mod` would work (although I've never tried it with a folder).

But in any case, I'd opt for the hardcoded version over an inappropriate path that needs to change. All it does is remove one place you have to upgrade the version for, and we'll need to change it later anyway.

Reply:
Actually, I'm just going to fix the location of go.mod. A single go.mod dependency stack at the root is almost always best practice. There are rare occasions where multiple go.mod stacks are fine within a single repo, but you still need to have a root go.mod in those cases. Trying to create isolated, independent go stacks within a single repo is not a good idea.

And I'm renaming `dockerfile` => `Dockerfile` ... 😞